### DISCUSSION TITLE
what is the recommendation with kubernetes?

### DISCUSSION THREAD

**Discussion by ktdreyer** [rpm-software-management/mock#1411] [Q&A]
My teammate @andrewschoen is trying to build RPMs in Tekton in IBM Cloud. `mock` errors out with `ERROR: Namespace unshare failed.` and `INFO: Guessed host environment type: unknown`

Is `mock` the best way to build RPMs in Tekton? Is this a future direction for `mock`? If not, what would you recommend instead?

**Comment by xsuchy** [comment]
See https://rpm-software-management.github.io/mock/#mock-inside-podman-fedora-toolbox-or-docker-container

**Comment by praiskup** [comment]
Mock runs fine in --privileged rootless Pod/container (with user_namespaces = root inside, normal user outside), which is likely not available in Tekton? The [experiment we have](https://github.com/konflux-fedora-rpm-packages/libecpg/blob/main/.tekton/libecpg-push.yaml#L24) allocates an external VM, and uses ssh to perform the build on that VM.

**Comment by andrewschoen** [comment]
I was able to run mock in a kubernetes privileged sidecar by running mock inside of another privilieged container started in the sidecar. I did something like this on the sidecar, where the `build-mock.sh` script is installing, configuring and running the mock command.

```
cat build-mock.sh | docker run -i --privileged quay.io/centos/centos:stream9 /bin/bash -
```
